modules/validation.rb

```diff
@@ -0,0 +1,30 @@
+require 'errors/error_class_type'
+require 'errors/error_empty_string'
+require 'errors/error_positive_value'
+
+module Validation
+  def validate(variables)
+    variables.each do |variable, class_type|
+      validate_class(variable, class_type)
+
+      case variable
+      when Integer
+        validate_integer_value(variable)
+      when String
+        validate_empty_string(variable)
+      end
```

Comment:
maybe move to separate methods? so this one won't be so overloaded
def validate_str(val)
def validate_int(val)